tests/workflows/map_node_state_gc/workflow.py

```diff
@@ -0,0 +1,53 @@
+from typing import List
+
+from vellum.workflows import BaseWorkflow
+from vellum.workflows.inputs.base import BaseInputs
+from vellum.workflows.nodes import MapNode
+from vellum.workflows.nodes.bases import BaseNode
+from vellum.workflows.outputs import BaseOutputs
+from vellum.workflows.state import BaseState
+
+
+class Inputs(BaseInputs):
+    items: List[int]
+
+
+class State(BaseState):
+    accumulated_data: List[str] = []
+
+
+class IterationNode(BaseNode):
+    item = MapNode.SubworkflowInputs.item
+    index = MapNode.SubworkflowInputs.index
+
+    class Outputs(BaseOutputs):
+        result: str
+        state_size: int
+
+    def run(self) -> Outputs:
+        large_string = f"iteration_{self.index}_" + ("x" * 1000)
+
+        self.state.accumulated_data.append(large_string)
```

Comment:
this is not accurately calculating memory usage.

First, we want to have some sort of memory measure like this

<img width="817" height="983" alt="Image" src="https://github.com/user-attachments/assets/4d0a7898-0ac5-40f2-844f-9a9403e7a9d8" />

Then we want Map node to have a max concurrency of like 4 or less.

Then, we want to make sure that the memory use across the duration of the MapNode is more or less staying the same and not increasing by some larger than expected amount on each iteration. We expect _some_ increase bc of the root node, but we want to make sure that the nested states are no longer in memory.

Towards the end of the test, we could try to see if the nested states are still accessible, or if they have been properly GC'd